src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -0,0 +1,191 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.ClassificationCategory;
+import com.google.cloud.language.v1.ClassifyTextRequest;
+import com.google.cloud.language.v1.ClassifyTextResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.language.v1.EntityMention;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.LanguageServiceSettings;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/*
+* This class creates objects that analyse text by 
+* extracting the categories, the mood, the events.
+* The scope is to get a final set of the key words
+* from the text. 
+*/
+
+public final class TextAnalyser {
+  private final String message;
+  private static final double EPSILON = 0.00001;
+
+  public TextAnalyser(String message) {
+     this.message = message.toLowerCase();
+  }
+
+  public final LanguageServiceSettings getSettings() throws IOException {
+    HeaderProvider headerProvider =
+            FixedHeaderProvider.create("X-Goog-User-Project","google.com:gpostcard");
+    return LanguageServiceSettings.newBuilder()
+                .setHeaderProvider(headerProvider)
+                .build();
+  }
+
+  public float getSentimentScore() throws IOException {
+    try (LanguageServiceClient languageService = LanguageServiceClient.create(getSettings())) {
```

Comment:
I never got around to use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html, I don't think that was available to me when I was more actively writing Java. Thanks for teaching me something here, I didn't even know it exists now.